### PR TITLE
Logging correlation with trace context

### PR DESCRIPTION
**Pull request by reyang**
Adding trace context logging correlation https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/logs/overview.md#log-correlation.

Baggage and Resource are not covered in this PR.

### CODE REVIEW

File: src/OpenTelemetry/Logs/LogRecord.cs [hunk 1]
```diff
@@ -28,6 +29,16 @@ public sealed class LogRecord
         internal LogRecord(DateTime timestamp, string categoryName, LogLevel logLevel, EventId eventId, object state, Exception exception)
         {
             this.Timestamp = timestamp;
+
+            var activity = Activity.Current;
```
Comment (cijothomas) [inline src/OpenTelemetry/Logs/LogRecord.cs:33]: should we check if this information is available in Scope (starting .net 5.0 and if enabled explicitly), and if not, then fall back to Activity.Current?

Comment (reyang) [reply src/OpenTelemetry/Logs/LogRecord.cs:33]: Probably no as this will be faster?

Comment (cijothomas) [reply src/OpenTelemetry/Logs/LogRecord.cs:33]: mm. I suppose this is fine then.

We can remove this section as well.
https://github.com/open-telemetry/opentelemetry-dotnet/tree/master/docs/logs/correlation#correlate-logs-with-traces

(That makes me wonder - what is the use case of Ilogger providing this feature in 5.0, as every provider can simply access Activity.Current if they chose to, right?)

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry/Logs/LogRecord.cs (hunk 1 above)
Comment (reyang) [reply src/OpenTelemetry/Logs/LogRecord.cs:33]: I think it'll benefit existing providers which already support scope.

**Pull request merged**